src/python/pants/core/goals/lint.py

```diff
@@ -282,58 +415,50 @@ async def lint(
     union_membership: UnionMembership,
     dist_dir: DistDir,
 ) -> Lint:
```

Comment:
So, this `@rule` has gotten pretty long. A few options:
1. Add a docstring that explains the big picture of what is happening.
2. Add more comments to various sections to highlight the steps.
3. Break out some `@_rule_helpers` with more self-explanatory names.

Doing one of those here would probably be good before landing, but not a blocker. It's possible that breaking out `@_rule_helpers` would be easier to review as a followup.

Reply:
FWIW part of this escapade is to simplify this. Mind if I punt and we revisit after the string of refactors? That way we don't make more work for ourselves.